x-pack/plugins/cases/server/common/limiter_checker/limiters/persistableStateAndExternalReferences.test.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createAttachmentServiceMock } from '../../../services/mocks';
+import { PersistableStateAndExternalReferencesLimiter } from './persistableStateAndExternalReferences';
+import {
+  createExternalReferenceRequests,
+  createFileRequests,
+  createPersistableStateRequests,
+  createUserRequests,
+} from '../test_utils';
+import { MAX_PERSISTABLE_STATE_AND_EXTERNAL_REFERENCES } from '../../../../common/constants';
+
+describe('PersistableStateAndExternalReferencesLimiter', () => {
+  const caseId = 'test-id';
+  const attachmentService = createAttachmentServiceMock();
+  attachmentService.countPersistableStateAndExternalReferenceAttachments.mockImplementation(
+    async () => {
+      return 1;
+    }
+  );
+
+  const limiter = new PersistableStateAndExternalReferencesLimiter(attachmentService);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('public fields', () => {
+    it('sets the errorMessage to the 1k limit', () => {
+      expect(limiter.errorMessage).toMatchInlineSnapshot(
+        `"Case has reached the maximum allowed number (100) of attached persistable state and external reference attachments."`
+      );
+    });
+
+    it('sets the limit to 1k', () => {
+      expect(limiter.limit).toBe(MAX_PERSISTABLE_STATE_AND_EXTERNAL_REFERENCES);
+    });
+  });
+
+  describe('countOfItemsWithinCase', () => {
+    it('calls the attachment service with the right params', () => {
+      limiter.countOfItemsWithinCase(caseId);
+
+      expect(
+        attachmentService.countPersistableStateAndExternalReferenceAttachments
+      ).toHaveBeenCalledWith({ caseId });
+    });
+  });
+
+  describe('countOfItemsInRequest', () => {
+    it('returns 0 when passed an empty array', () => {
+      expect(limiter.countOfItemsInRequest([])).toBe(0);
+    });
+
+    it('returns 0 when the requests are not for persistable state attachments or external references', () => {
+      expect(limiter.countOfItemsInRequest(createUserRequests(2))).toBe(0);
+    });
+
+    it('counts persistable state attachments or external references correctly', () => {
+      expect(
+        limiter.countOfItemsInRequest([
+          createPersistableStateRequests(1)[0],
```

Comment:
nit: Could you please add a `createUserRequests` and `createFileRequests` in the collection?